packages/leancode_hooks/lib/src/use_decarative_text_editing_controller.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+TextEditingController useDeclarativeTextEditingController({
+  required String text,
+}) {
+  final controller = useTextEditingController(text: text);
+
+  useEffect(
+    () {
+      if (controller.text != text) {
+        WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+          controller.text = text;
+        });
+      }
+      return null;
+    },
+    [text],
+  );
+
+  return controller;
+}
```

Comment:
@shilangyu Maybe we could merge this somehow with mine `useSyncedTextEditingController`?

I thought of using `useDeclarativeTextEditingController` inside `useSyncedTextEditingController`.e

Reply:
Would you like to refactor `useDeclarativeTextEditingController` to also handle these edge cases or should we ditch it?

 To me, it looks like a useful hook.

Reply:
TBH I dont understand what `useSyncedTextEditingController` does, looks like it only adds a callback. I though it would do what `useDeclarativeTextEditingController` aimed to do

Reply:
It's basically a common hook wrapping `useTextEditingController` with adding a listener:

I found myself writing this quite often
```dart
  final controller = useTextEditingController(text: initialText);

  void listener() => onChanged(controller.value);

  useEffect(
    () {
      controller.addListener(listener);
      return () => controller.removeListener(listener);
    },
    [controller, onChanged],
  );

  return controller;
```

so I extracted this to a hook.

I'd say that for `v0.1.0` we can keep this.